lively.morphic/morph.js

```diff
@@ -1696,10 +1701,16 @@ export class Morph {
     return this;
   }
 
+  /**
+   * Different than `remove`, this method should be used to indicate than this morph is no longer needed.
+   * It can be handy to e.g. clean up connections and other resources.
+   * @param {Boolean} remove - Whether or not remove() should be called. Normally, it does not make sense to set this to be false, as abandoning indicates that a morph is no longer to be used.
+   * `remove` was introduced as a band-aid for the old master component system (before we had declarative components)
+   * FIXME: can the parameter be removed?
+   */
   abandon (remove = true) {
-    // Use this method to signal the wish to permanently delete this object. Overwrite this method to clean up resources on its deletion. We do not have access to the JS VM, this method does not interact with the garbage collector and does not result in the actual deletion of the object, if there are still left over references to this object.
     signal(this, 'onAbandon');
-    this.emptyComments();
+    if ($world.isIDEWorld) $world.emptyCommentsFor(this);
```

Comment:
Here `$world` is definitely referencing itself right? Like we could also replace `$world` with `this`?

Reply:
I don't think so? We are in `Morph` here...